src/averageAges.js

```diff
@@ -55,7 +63,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const hasMother = people.filter(child => {
+    const mother = people.find(person => person.name === child.mother);
```

Comment:
You are using a 'find' method inside a 'filter' method which leads to unnecessary computations and may slow down the execution of the program for large datasets. It is better to create a mapping of mothers to children before the 'filter' method to avoid this.